src/ProcessTable.cpp

```diff
@@ -0,0 +1,165 @@
+#include "ProcessTable.h"
+
+#include <algorithm>
+#include <cctype>
+#include <chrono>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <thread>
+#include <unordered_map>
+
+#include "SystemInfo.h"
+
+std::ostream& operator<<(std::ostream& os, const ProcessState& state) {
+  switch (state) {
+    case ProcessState::Running:
+      return os << "Running";
+    case ProcessState::Sleeping:
+      return os << "Sleeping";
+    case ProcessState::DiskSleep:
+      return os << "Disk Sleep";
+    case ProcessState::Stopped:
+      return os << "Stopped";
+    case ProcessState::Zombie:
+      return os << "Zombie";
+    case ProcessState::Dead:
+      return os << "Dead";
+    case ProcessState::Idle:
+      return os << "Idle";
+    default:
+      return os << "Unknown";
+  }
+}
+
+std::ostream& operator<<(std::ostream& os, const ProcessInfo& info) {
+  os << std::left << std::setw(8) << info.pid << std::setw(40) << info.name
+     << std::setw(10) << info.state << std::setw(6) << std::fixed
+     << std::setprecision(2) << info.cpuUsed << std::setw(10) << info.memUsedKB;
+  return os;
+}
+
+void ProcessTable::printTableHeader() const {
+  std::cout << std::left << std::setw(8) << "PID" << std::setw(40) << "Name"
+            << std::setw(10) << "State" << std::setw(6) << "% CPU"
+            << std::setw(10) << "RAM KB" << "\n";
+}
+
+namespace fs = std::filesystem;
+
+static bool isNumber(const std::string& s) {
```

Comment:
static для функции - это способ сделать внутреннюю линковку в C стиле

в C++ для этого используем анонимное пространство имен (без названия)

```suggestion
namespace {

bool isNumber(const std::string& s) { ... }

}
```